dotnet/samples/Demos/ModelContextProtocolClientServer/MCPClient/Program.cs

```diff
@@ -18,21 +17,68 @@ namespace MCPClient;
+        await using IMcpClient mcpClient = await CreateMcpClientAsync();
+
+        // Retrieve and display the list provided by the MCP server
+        IList<McpClientTool> tools = await mcpClient.ListToolsAsync();
+        DisplayTools(tools);
+
+        // Create a kernel and register the MCP tools
+        Kernel kernel = CreateKernelWithChatCompletionService();
+        kernel.Plugins.AddFromFunctions("Tools", tools.Select(aiFunction => aiFunction.AsKernelFunction()));
+
+        // Enable automatic function calling
+        OpenAIPromptExecutionSettings executionSettings = new()
+        {
+            Temperature = 0,
+            FunctionChoiceBehavior = FunctionChoiceBehavior.Auto(options: new() { RetainArgumentTypes = true })
+        };
+
+        string prompt = "What is the likely color of the sky in Boston today?";
+
+        // Execute a prompt using the MCP tools. The AI model will automatically call the appropriate MCP tools to answer the prompt.
+        FunctionResult result = await kernel.InvokePromptAsync(prompt, new(executionSettings));
+
+        Console.WriteLine(result);
+
+        // The expected output is: The likely color of the sky in Boston today is gray, as it is currently rainy.
+    }
+
+    /// <summary>
+    /// Creates an instance of <see cref="Kernel"/> with the OpenAI chat completion service registered.
+    /// </summary>
+    /// <returns>An instance of <see cref="Kernel"/>.</returns>
+    private static Kernel CreateKernelWithChatCompletionService()
     {
         // Load and validate configuration
-        var config = new ConfigurationBuilder()
+        IConfigurationRoot config = new ConfigurationBuilder()
             .AddUserSecrets<Program>()
             .AddEnvironmentVariables()
             .Build();
 
         if (config["OpenAI:ApiKey"] is not { } apiKey)
         {
-            Console.Error.WriteLine("Please provide a valid OpenAI:ApiKey to run this sample. See the associated README.md for more details.");
-            return;
+            string message = "Please provide a valid OpenAI:ApiKey to run this sample. See the associated README.md for more details.";
+            Console.Error.WriteLine(message);
+            throw new InvalidOperationException(message);
         }
 
-        // Create an MCP client
-        await using var mcpClient = await McpClientFactory.CreateAsync(
+        // Create kernel
+        var kernelBuilder = Kernel.CreateBuilder();
+        kernelBuilder.Services.AddOpenAIChatCompletion(serviceId: "openai", modelId: config["OpenAI:ChatModelId"] ?? "gpt-4o-mini", apiKey: config["OpenAI:ApiKey"]!);
+
+        return kernelBuilder.Build();
+    }
+
+    /// <summary>
+    /// Creates an MCP client and connects it to the MCPServer server.
+    /// </summary>
+    /// <returns>An instance of <see cref="IMcpClient"/>.</returns>
+    private static async Task<IMcpClient> CreateMcpClientAsync()
+    {
+        var mcpClient = await McpClientFactory.CreateAsync(
```

Comment:
Or just use directly the McpClientFactory.CreateAsync.

```suggestion
    private static Task<IMcpClient> CreateMcpClientAsync()
        => McpClientFactory.CreateAsync(
```